grpc/binlog/v1alpha/binarylog.proto

```diff
@@ -52,7 +52,13 @@ message GrpcLogEntry {
   // The logger uses one of the following fields to record the payload,
   // according to the type of the log entry.
   oneof payload {
-    // Used by CLIENT_INIT_METADATA, SERVER_INIT_METADATA and TRAILING_METADATA
+    // Used by CLIENT_INIT_METADATA, SERVER_INIT_METADATA and
```

Comment:
Why don't we just say that all implementations will log it as part of metadata?  We should not have the format be different across languages.

Reply:
In Java, things like ":path" do not appear in the metadata object shown to applications. I believe the motivation was to hide http/2 specific implementation details from the interceptors API. This info instead copied into the [io.grpc.MethodDescriptor](https://grpc.io/grpc-java/javadoc/io/grpc/MethodDescriptor.html#getFullMethodName--). One of the things a11r mentioned about binlog is that it should be high level and do what a user application could have done, which is why I am using public APIs for logging the data.

Reply:
To expand on this, this is especially true when we're talking about in-process transports or transport types we have internally. The status, method, and other special information that is always present may be special cased and not actually put into the metadata. So maybe "language differences" is a misnomer here.

Reply:
I'm fine with the idea that we don't want this encoded in the metadata because that's not appropriate for all transports.  But we should not do this differently by language.  If we are going to introduce new fields for this information, then those fields should be used by all languages.

Reply:
Updated the proto so that gRPC information is captured as top level fields, rather than relying an implementation that puts them as fields of the Metadata.

Metadata is still a place where things can diverge, because right now C is logging it after things like `:path` are added.  Do we want to filter out this duplicate info to be consistent? I think the place where go might log is high level and would omit those fields as well. @dfawley is that correct?

Reply:
+1 on that

Reply:
In that case, please remove the part of this comment that says that the values differ based on the transport implementation.

Reply:
It is not clear to me what is meant by "added by grpc library itself" as there could be filters putting in metadata as needed and some of the information could be useful?

Reply:
I think if the filter is something that is user accessible then it's fair to log it. But if the filter is a part of gRPC's internals then it's fine to omit that data, since the binary log should be capturing what the user application is doing.

Reply:
Since the intent here is to make binary logging be close to the application level rather than close to the wire level, I think it's fine to not log anything that is removed before the metadata is returned to the application.